presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -1006,7 +1013,16 @@ public long getMaxActiveSplitTime()
     {
         Iterator<RunningSplitInfo> iterator = runningSplitInfos.iterator();
         if (iterator.hasNext()) {
-            return NANOSECONDS.toMillis(ticker.read() - iterator.next().getStartTime());
+            RunningSplitInfo splitInfo = iterator.next();
+            long duration = NANOSECONDS.toMillis(ticker.read() - splitInfo.getStartTime());
+            if (duration > 1_000_000) {
```

Comment:
This defeats the purpose of using `TimeUnit`. Convert directly to the target unit of comparison:
```java
long seconds = NANOSECONDS.toMillis(...);
```

However, it's better to use `Duration`:
```java
Duration duration = succinctNanos(ticker.read() - splitInfo.getStartTime());
if (duration.getValue(SECONDS) >= 1000) {
```

Even better, add a constant and use that:
```java
if (duration.compareTo(LONG_SPLIT_WARNING_THRESHOLD) >= 0) {
```